vlemish/3SumSolution.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace _3Sum
+{
+    public class _3SumSolution
+    {
+        // The easiest way to solve the problem is to decrease problem to 2Sum problem
+        // we can do it by creating a dictionary with elements for which we're going to search further.
+        public IList<IList<int>> FindTriples(int[] nums)
+        {
+            // if length is less than 3 -> no triples can be found
+            if (nums.Length < 3)
+            {
+                return new List<IList<int>>();
+            }
+
+            Array.Sort(nums);
+
+            List<Tripple> tripples = new List<Tripple>();
+
+            // create dictionary with elements to search
+            Dictionary<int, int> numsDictionary = new();
+
+            // use pointers to decrease number of iterations
+            // take the first element and add it to dictionary and then use this element to find couple of elements which will give us 0
+            for (int i = 1; i < nums.Length - 1; i++)
```

Comment:
Strange to see you start loop with 1 but then assign k = i - 1 to make it zero. Why not go the other way around.. just interesting

Reply:
`K` was an element that will be added to Dictionary and then used to determine whether there is an element in a dictionary which will be equal to (i+j ) * -1 and if so, then the triplet is found. I could go another way by adding `i` element  to the dictionary, and having k = i+1; j = k+1; But it changes nothing, I guess. For me it looked simpler to understand what's going on :)